tests/publiccloud/registration.pm

```diff
@@ -36,7 +36,15 @@ sub run {
 
 sub cleanup {
     my ($self) = @_;
-    $self->{instance}->upload_log('/var/log/cloudregister', log_name => $autotest::current_test->{name} . '-cloudregister.log');
+    if ($self->{result} =~ /fail$/i) {
```

Comment:
you don't need to check for failure here , this logic is handled "above" . In this dimension the rule of thumb  if this function was called you need to upload all logs

Reply:
In the PR Acceptance Criteria you asked explicitly those logs for only fail case. I worked in this sense.

PLease, let me know if ticket shall be updated and PR too for *always add all logs* 🙂

Reply:
Given above request, I'm going to re-update code, removing the fail-condition so all the 4 logs will be collected unconditionally.